versioned/tests/src/main/java/org/projectnessie/versioned/tests/AbstractCommitLog.java

```diff
@@ -144,24 +146,23 @@ public void commitLogExtendedNoGlobalState() throws Exception {
         .containsExactlyElementsOf(hashes);
 
     List<Commit> commits = Lists.reverse(commitsList(branch, true));
-    soft.assertThat(IntStream.rangeClosed(1, numCommits))
-        .allSatisfy(
-            i -> {
-              Commit c = commits.get(i - 1);
-              assertThat(c)
-                  .extracting(
-                      Commit::getCommitMeta,
-                      Commit::getHash,
-                      Commit::getParentHash,
-                      Commit::getOperations)
-                  .containsExactly(
-                      CommitMeta.fromMessage("Commit #" + i),
-                      hashes.get(i - 1),
-                      parentHashes.get(i - 1),
-                      Arrays.asList(
-                          Delete.of(Key.of("delete" + i)),
-                          Put.of(Key.of("k" + i), onRef("v" + i, "c" + i)),
-                          Put.of(Key.of("key" + i), onRef("value" + i, "cid" + i))));
-            });
+    for (int i = 1; i <= numCommits; i++) {
```

Comment:
Why not validate the "initial" commit too (i.e. at index `0`)?.. Maybe in outside this loop, though.

Reply:
Let's not make this one too complicated.

Reply:
This one is a nit (optional), I agree, but how about `subList` above? I think it makes sense to remove it now since the intention is to process the whole list, right?

Reply:
Still I find having `firstParent` in `parentHashes` confusing if we do not validate it... might as well use `null`... Sorry about nitpicking, just hoping for a more readable code :)